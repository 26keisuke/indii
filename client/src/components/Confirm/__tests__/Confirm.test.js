// import React from "react"
// import { mount } from "enzyme"
// import { Confirm } from "../Confirm"

// function init(next) {
//     return {
//         update: {
//             confirmation: {
//                 on: true,
//                 id: "TEST_ID",
//                 action: "TEST_ACTION",
//                 title: "TEST_TITLE",
//                 message: "TEST_MESSAGE",
//                 buttonMessage: "TEST_BTN",
//                 next: next,
//                 value: "TEST_VALUE"
//             }
//         }
//     }
// }

// var action;

// const postAction = (shouldProceed) => {
//     action = shouldProceed
// }

// const props = init()
// const wrapper = mount(<Confirm {...props} postAction={postAction}/>)

// const withNext = (nextName) => {
//     const props = init(nextName)

//     return {
//         wrapperWithNext: mount(<Confirm {...props} postAction={postAction}/>),
//         props: props
//     }
// }

// describe("Confirm", () => {
//     it("rendered w/o error", () => {
//         expect(wrapper).toMatchSnapshot()
//     })

//     describe("correct text", () => {
//         it("<Title>", () => {
//             const title = wrapper.find("h2")
//             expect(title.text()).toBe("TEST_TITLE")
//         })
    
//         it("<Message>", () => {
//             const message = wrapper.find("h4")
//             expect(message.text()).toBe("TEST_MESSAGE")
//         })
    
//         it("<Button>", () => {
//             const firstBtn = wrapper.find("button").at(0)
//             expect(firstBtn.text()).toBe("TEST_BTN")
//         })
//     })

//     describe("click events working correctly", () => {

//         beforeEach(() => {
//             action = undefined;
//         })

//         it("<Done>", () => {
//             const firstBtn = wrapper.find("button").at(0)
//             firstBtn.simulate("click")
//             expect(action).toBe(true)
//         })

//         it("<Next>", () => {
//             const { wrapperWithNext } = withNext("TEST_NEXT")
//             const firstBtn = wrapperWithNext.find("button").at(0)
//             firstBtn.simulate("click")
//             expect(action).toBe(undefined)
//         })

//         it("<Cancel(button)>", () => {
//             const secondBtn = wrapper.find("button").at(1)
//             secondBtn.simulate("click")
//             expect(action).toBe(false)
//         })

//         it("<Cancel(icon)>", () => {
//             const crossIcon = wrapper.find("svg")
//             crossIcon.simulate("click")
//             expect(action).toBe(false)
//         })
//     })
// })
