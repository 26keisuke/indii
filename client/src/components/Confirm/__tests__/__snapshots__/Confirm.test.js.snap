// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm is rendered w/o error 1`] = `
<Confirm
  postAction={[Function]}
  update={
    Object {
      "confirmation": Object {
        "action": "TEST_ACTION",
        "buttonMessage": "TEST_BTN",
        "id": "TEST_ID",
        "message": "TEST_MESSAGE",
        "next": undefined,
        "on": true,
        "title": "TEST_TITLE",
        "value": "TEST_VALUE",
      },
    }
  }
>
  <Fade
    in={true}
  >
    <Transition
      appear={false}
      enter={true}
      exit={true}
      in={true}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      timeout={100}
      unmountOnExit={false}
    >
      <styled.div
        style={
          Object {
            "opacity": 1,
            "top": "50%",
            "transition": "all 100ms ease-in-out",
          }
        }
      >
        <div
          className="sc-fzomuh RfEQn"
          style={
            Object {
              "opacity": 1,
              "top": "50%",
              "transition": "all 100ms ease-in-out",
            }
          }
        >
          <div>
            <Styled(IoMdClose)
              onClick={[Function]}
            >
              <IoMdClose
                className="sc-fzoJus dkKAUY"
                onClick={[Function]}
              >
                <IconBase
                  attr={
                    Object {
                      "viewBox": "0 0 512 512",
                    }
                  }
                  className="sc-fzoJus dkKAUY"
                  onClick={[Function]}
                >
                  <svg
                    className="sc-fzoJus dkKAUY"
                    fill="currentColor"
                    height="1em"
                    onClick={[Function]}
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M405 136.798L375.202 107 256 226.202 136.798 107 107 136.798 226.202 256 107 375.202 136.798 405 256 285.798 375.202 405 405 375.202 285.798 256z"
                      key="0"
                    />
                  </svg>
                </IconBase>
              </IoMdClose>
            </Styled(IoMdClose)>
            <styled.h2>
              <h2
                className="sc-fzqAui dEKnBj"
              >
                TEST_TITLE
              </h2>
            </styled.h2>
            <styled.h4>
              <h4
                className="sc-fzqMdD djoGIr"
              >
                TEST_MESSAGE
              </h4>
            </styled.h4>
            <styled.p>
              <p
                className="sc-fzowVh sgLwX"
              />
            </styled.p>
            <styled.div>
              <div
                className="sc-fzoaKM kRgDba"
              >
                <button
                  onClick={[Function]}
                >
                  TEST_BTN
                </button>
                <button
                  onClick={[Function]}
                >
                  キャンセル
                </button>
              </div>
            </styled.div>
          </div>
        </div>
      </styled.div>
    </Transition>
  </Fade>
</Confirm>
`;

exports[`Confirm rendered w/o error 1`] = `
<Confirm
  postAction={[Function]}
  update={
    Object {
      "confirmation": Object {
        "action": "TEST_ACTION",
        "buttonMessage": "TEST_BTN",
        "id": "TEST_ID",
        "message": "TEST_MESSAGE",
        "next": undefined,
        "on": true,
        "title": "TEST_TITLE",
        "value": "TEST_VALUE",
      },
    }
  }
>
  <Fade
    in={true}
  >
    <Transition
      appear={false}
      enter={true}
      exit={true}
      in={true}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      timeout={100}
      unmountOnExit={false}
    >
      <styled.div
        style={
          Object {
            "opacity": 1,
            "top": "50%",
            "transition": "all 100ms ease-in-out",
          }
        }
      >
        <div
          className="sc-fzomuh RfEQn"
          style={
            Object {
              "opacity": 1,
              "top": "50%",
              "transition": "all 100ms ease-in-out",
            }
          }
        >
          <div>
            <Styled(IoMdClose)
              onClick={[Function]}
            >
              <IoMdClose
                className="sc-fzoJus dkKAUY"
                onClick={[Function]}
              >
                <IconBase
                  attr={
                    Object {
                      "viewBox": "0 0 512 512",
                    }
                  }
                  className="sc-fzoJus dkKAUY"
                  onClick={[Function]}
                >
                  <svg
                    className="sc-fzoJus dkKAUY"
                    fill="currentColor"
                    height="1em"
                    onClick={[Function]}
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M405 136.798L375.202 107 256 226.202 136.798 107 107 136.798 226.202 256 107 375.202 136.798 405 256 285.798 375.202 405 405 375.202 285.798 256z"
                      key="0"
                    />
                  </svg>
                </IconBase>
              </IoMdClose>
            </Styled(IoMdClose)>
            <styled.h2>
              <h2
                className="sc-fzqAui dEKnBj"
              >
                TEST_TITLE
              </h2>
            </styled.h2>
            <styled.h4>
              <h4
                className="sc-fzqMdD djoGIr"
              >
                TEST_MESSAGE
              </h4>
            </styled.h4>
            <styled.p>
              <p
                className="sc-fzowVh sgLwX"
              />
            </styled.p>
            <styled.div>
              <div
                className="sc-fzoaKM kRgDba"
              >
                <button
                  onClick={[Function]}
                >
                  TEST_BTN
                </button>
                <button
                  onClick={[Function]}
                >
                  キャンセル
                </button>
              </div>
            </styled.div>
          </div>
        </div>
      </styled.div>
    </Transition>
  </Fade>
</Confirm>
`;
